config/config.go

```diff
@@ -260,15 +260,21 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	// Do global overrides and validate unique names.
 	jobNames := map[string]struct{}{}
 	for _, scfg := range c.ScrapeConfigs {
+		// First set the correct scrape interval, then check whether the timeout
```

Comment:
This reads oddly. Either "check whether the timeout is less than that" or "check that the timeout is not greater than that".

Same below.